File: C#/C# 1/3. Operators-Expressions-Statements-Homework/Prime-Check/PrimeCheck.cs
/* Problem 8. Prime Check
   Write a program that reads an integer N (which will always be less than 100 or equal) and uses an expression to check if given N is prime (i.e. it is divisible without remainder only to itself and 1). Note: You should check if the number is positive. Output true if the number is prime and false otherwise. */

using System;

class PrimeCheck
{
    static void Main()
    {
        int posIntNumber = int.Parse(Console.ReadLine());

        if (posIntNumber < 2)  // when posIntNumber < 0 or posIntNumber = 0 or posIntNumber = 1
        {
            Console.WriteLine("false");
            return;
        }

        bool boolPrime = true;
        if ((posIntNumber == 2) || (posIntNumber == 5))
        {
            boolPrime = true;
        }
        else
        {
            // (posIntNumber <= 1) || (posIntNumber % 2 == 0) || (posIntNumber % 5 == 0)
            if ((posIntNumber % 2 == 0) || (posIntNumber % 5 == 0))                 
            {
                boolPrime = false;
            }
            else
            {

                int maxDivider = (int)Math.Sqrt(posIntNumber);                
                for (int i = 3; i <= maxDivider; i++)
                {
                    if (posIntNumber % i == 0)
                    {
                        boolPrime = false;
                        break;
                    }
                }
                
                //// other way
                //int maxDivider = (int)Math.Sqrt(posIntNumber);
                //int count = 0;
                //for (int i = 3; i <= maxDivider; i++)
                //{
                //    if (posIntNumber % i == 0)
                //    {
                //        count++;
                //    }
                //}
                //if (count > 0)
                //{
                //    boolPrime = false;
                //}
            }
        }
        Console.WriteLine(boolPrime ? "true" : "false");
    }
}

File: Front-end/JS Fundamentals/Operators Аnd Expressions/07. Prime Check/prime-check.js
function solve(args) {
    var num = +args[0];

    if (num < 2) { // num < 0 or num === 0 or num === 1
        console.log('false');
        return;
    }

    var isPrime = true;

    if (num === 2 || num === 5) {
        isPrime = true;
    } else {
        if (num % 2 === 0 || num % 5 === 0) {
            isPrime = false;
        } else {
            var maxDivider = Math.sqrt(num);

            for (var i = 3; i <= maxDivider; i += 1) {
                if (num % i === 0) {
                    isPrime = false;
                    break;
                }
            }
        }
    }

    console.log(isPrime);
}

solve([2]);
solve([23]);
solve([-3]);
solve([0]);
solve([1]);